flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionType.java

```diff
@@ -49,13 +59,17 @@
 	/** Does this partition use a limited number of (network) buffers? */
 	private final boolean isBounded;
 
+	/** This partition will not be released after consuming if 'isPersistent' is true. */
+	private final boolean isPersistent;
```

Comment:
I think we want to introduce a property to indicate this partition is global across jobs. So `isPersistent` seems not reflect this behavior, because `BLOCKING` type is also for persistent file from the semantics.  Maybe called `isGlobal`?

Reply:
I think `BLOCKING ` is opposite to `PIPELINE` only describes the flow pattern of data.
But persistence describes the life cycle of data.
What do you think?

Reply:
We want to define a property to give a hint for partition release. But the term of `isPersistent` seems not very proper for this property, and it looks like to describe whether the data is saved in memory or persistent file. So from this point, `BLOCKING` also saves the data in persistent file. I think `isGlobal` seems more suitable for the hint of partition release.

Reply:
How about changing `BLOCKING` to `BLOCKING_INTERNAL` and `BLOCKING_PERSISTENT` to `BLOCKING_EXTERNAL`

Reply:
I agree that BLOCKING_PERSISTENT is not the best name.
The specific thing is that it lives across jobs, but I am not sure what a good name for that would be.
"global" and "external" both go in that direction...

Reply:
It seems "external" might be better than "global" since "internal" is better than "local"?